src/libraries/System.Data.Common/ref/System.Data.Common.cs

```diff
@@ -1162,15 +1164,18 @@ public partial interface IDataAdapter
         System.Data.IDataParameter[] GetFillParameters();
         int Update(System.Data.DataSet dataSet);
     }
+#nullable enable
     public partial interface IDataParameter
     {
         System.Data.DbType DbType { get; set; }
         System.Data.ParameterDirection Direction { get; set; }
         bool IsNullable { get; }
+        [System.Diagnostics.CodeAnalysis.AllowNull]
         string ParameterName { get; set; }
+        [System.Diagnostics.CodeAnalysis.AllowNull]
         string SourceColumn { get; set; }
         System.Data.DataRowVersion SourceVersion { get; set; }
-        object Value { get; set; }
+        object? Value { get; set; }
```

Comment:
`IDataParameter.Value` can be null... should `IDataParameterCollection[string]` below return `object?` instead of `object`?

Reply:
`IDataParameterCollection[string]` returns the DbParameter instance, not its value - and AFAIK it's never supposed to contain a null...

Reply:
Ah, ok.